### PR TITLE
BZ#2118790 - Convert2RHEL - Content views for activation keys

### PR DESCRIPTION
**Pull request by stejskalleos**
Add content views for Convert2RHEL activation keys
with enabled repositories only for the specific OS

**Steps to test**
- install centos7
- update host
- reboot host so that new kernel is loaded
- register host to satellite with "CentOS 7 converting" host group
- Run `yum install convert2rhel`

For more details see https://bugzilla.redhat.com/show_bug.cgi?id=2118790

### CODE REVIEW

File: roles/convert2rhel/tasks/activation_keys.yml [hunk 1]
```diff
@@ -31,8 +31,7 @@
     organization: "{{ foreman_organization }}"
     name: "{{ foreman_convert2rhel_key_centos7 }}"
     lifecycle_environment: "{{ foreman_convert2rhel_lifecycle_env }}"
-    content_view: "{{ foreman_convert2rhel_content_view }}"
-    subscriptions: "{{ foreman_convert2rhel_org_info['organization']['simple_content_access'] | ternary(omit, foreman_convert2rhel_centos7_subs) }}"
```
Comment (evgeni) [inline roles/convert2rhel/tasks/activation_keys.yml:35]: Are subscriptions unnecessary now? (I'd guess in the non-SCA case, they are still needed?)

File: roles/convert2rhel/tasks/activation_keys.yml [hunk 2]
```diff
@@ -31,8 +31,7 @@
     organization: "{{ foreman_organization }}"
     name: "{{ foreman_convert2rhel_key_centos7 }}"
     lifecycle_environment: "{{ foreman_convert2rhel_lifecycle_env }}"
-    content_view: "{{ foreman_convert2rhel_content_view }}"
```
Comment (evgeni) [inline roles/convert2rhel/tasks/activation_keys.yml:34]: This CV is still used in the `host_groups` playbook, is that correct?!

File: roles/convert2rhel/tasks/content_views.yml [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+  theforeman.foreman.content_view:
+    username: "{{ foreman_username }}"
+    password: "{{ foreman_password }}"
+    server_url: "{{ foreman_server_url }}"
+    validate_certs: "{{ foreman_validate_certs | default(omit) }}"
+    organization: "{{ foreman_organization }}"
+    name: "{{ foreman_convert2rhel_key_oracle7 }}"
+    repositories:
+      - name: '{{ foreman_convert2rhel_oracle7_repo }}'
+        product: '{{ foreman_convert2rhel_oracle7_product }}'
```
Comment (evgeni) [inline roles/convert2rhel/tasks/content_views.yml]: don't you need both, `foreman_convert2rhel_rhel7_repo` AND `foreman_convert2rhel_oracle7_repo`, in the Oracle case?

File: roles/convert2rhel/tasks/content_views.yml [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+  theforeman.foreman.content_view:
+    username: "{{ foreman_username }}"
+    password: "{{ foreman_password }}"
+    server_url: "{{ foreman_server_url }}"
+    validate_certs: "{{ foreman_validate_certs | default(omit) }}"
+    organization: "{{ foreman_organization }}"
+    name: "{{ foreman_convert2rhel_key_oracle8 }}"
+    repositories:
+      - name: '{{ foreman_convert2rhel_oracle8_repo }}'
+        product: '{{ foreman_convert2rhel_oracle8_product }}'
```
Comment (evgeni) [inline roles/convert2rhel/tasks/content_views.yml]: don't you need both, `foreman_convert2rhel_rhel8_repo` AND `foreman_convert2rhel_oracle8_repo`, in the Oracle case?

**Comment by evgeni** [comment]
The tests seem to pass and all, but I am not sure about the actual changes and their influence on the c2r workflow.

Paging @ares as I don't have an environment to test this fully.

File: roles/convert2rhel/tasks/activation_keys.yml (hunk 2 above)
Comment (stejskalleos) [reply roles/convert2rhel/tasks/activation_keys.yml:34]: Yes it's correct. Now if I'm thinking about that, we should probably also change the CVs for the host groups, kinda forgot about it when was doing this PR

**Comment by stejskalleos** [comment]
@evgeni updated

* Changes only for CentOS 7 and CentOS 8
* There is still probably issue with Oracle but  I am unable to verify it due to various issues. Since the issue is reported for CentOS let's focus on CentOS only
* Removed content views and environments from the host groups. Registered host will get them from activation keys anyway

File: roles/convert2rhel/README.md [hunk 5]
```diff
@@ -36,7 +36,7 @@ Convert2RHEL
         foreman_manifest_path: "~/manifest.zip"
         foreman_content_rhel_enable_rhel7: true
         foreman_content_rhel_enable_rhel8: true
-        foreman_content_rhel_rhel8_releasever: 8.4
+        foreman_content_rhel_rhel8_releasever: 8.5
```
Comment (evgeni) [inline roles/convert2rhel/README.md:39]: Why not 8.6? ;)

Comment (stejskalleos) [reply roles/convert2rhel/README.md:39]: Because it doesn't work with `8.6`, see https://access.redhat.com/discussions/5676681

Comment (evgeni) [reply roles/convert2rhel/README.md:39]: wow.

**Comment by evgeni** [comment]
The changes look good to me from an Ansible PoV, but I don't have an environment to test the full workflow.

@stejskalleos @ares if you're confident this works, we can merge.

**Review by evgeni — APPROVED** [review]

**Comment by stejskalleos** [comment]
@ares are you ok with the changes? I'm confident it works, tested workflow with CentOS 7 & 8 and it worked.

**Review by ares — APPROVED** [review]
I think this is fine and can be merged.

**Pull request merged**